docs/user_guide/productionizing/configuring_access_to_gpus.md

```diff
@@ -6,31 +6,405 @@
-that explicitly request GPUs. To achieve this, Flyte uses the Kubernetes concept of [taints and tolerations](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/).
+Along with compute resources like CPU and memory, you may want to configure and access GPU resources. 
 
-Kubernetes can automatically apply tolerations for extended resources like GPUs using the [ExtendedResourceToleration plugin](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#extendedresourcetoleration), enabled by default in some cloud environments. Make sure the GPU nodes are tainted with a key matching the resource name, i.e., `key: nvidia.com/gpu`.
+Flyte provides different ways to request accelerator resources directly from the task decorator. This page covers the requirements and procedures to leverage them. 
 
-You can also configure Flyte backend to apply specific tolerations. This configuration is controlled under generic  k8s plugin configuration as can be found [here](https://github.com/flyteorg/flyteplugins/blob/5a00b19d88b93f9636410a41f81a73356a711482/go/tasks/pluginmachinery/flytek8s/config/config.go#L120).
+>The examples in this section use [ImageSpec](https://docs.flyte.org/en/latest/user_guide/customizing_dependencies/imagespec.html#imagespec), a Flyte feature that builds a custom container image without a Dockerfile. Install it using `pip install flytekitplugins-envd`.
 
-The idea of this configuration is that whenever a task that can execute on Kubernetes requests for GPUs, it automatically
-adds the matching toleration for that resource (in this case, `gpu`) to the generated PodSpec.
-As it follows here, you can configure it to access specific resources using the tolerations for all resources supported by
-Kubernetes.
+## Requesting a GPU with no preference for device
+The goal in this example is to run the task on a single available GPU :
 
-Here's an example configuration:
+```python
+from flytekit import ImageSpec, Resources, task
 
+image = ImageSpec(
+    base_image= "ghcr.io/flyteorg/flytekit:py3.10-1.10.2",
+     name="pytorch",
+     python_version="3.10",
+     packages=["torch"],
+     builder="envd",
+     registry="<YOUR_CONTAINER_REGISTRY>",
+ )
+
+@task(requests=Resources( gpu="1"))
+def gpu_available() -> bool:
+   return torch.cuda.is_available() # returns True if CUDA (provided by a GPU) is available
+```
+### How it works?
+
+![](https://raw.githubusercontent.com/flyteorg/static-resources/main/flyte/deployment/gpus/generic_gpu_access.png)
+
+When this task is evaluated, `flyteproller` injects a [toleration](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/) in the pod spec:
+
+```yaml
+tolerations:    nvidia.com/gpu:NoSchedule op=Exists
+```
+The Kubernetes scheduler will admit the pod if there are worker nodes in the cluster with a matching taint and available resources.
```

Comment:
When `accelerator` is not specified and there are multiple GPUs accelerator types configured, how does Kubernetes decide which accelerator to use?

Reply:
Great question. Currently K8s uses 3rd-party Device Driver plugins that advertise Extended Resources (this is, things other than CPU and memory) to the kubelet. In the case of NVIDIA, their GPU operator implements this approach using a DaemonSet that communicates with the K8s scheduler to let it know which devices are available. While it works, it's not great because there's a lot of coordination required between the K8s scheduler and this external driver, no good management of "distributed locking" for shared resources, and in consequence, Pods can be scheduled to a node but left in Pending state bc one of those Extended Resources is not ready.
This is changing with the introduction of Dynamic Resource Allocation (soon to be in beta on K8s 1.31) 

I just added [a brief note](https://github.com/flyteorg/flyte/pull/5515/files#diff-a0d7763c302fd0c5d6a029c8caa655ee9de0d6ce102da40dc8e2f793fe65f33eR45) about this but let me know if you think a better explanation is needed.